### DISCUSSION TITLE
Prior Match Weight Inconsistency

### DISCUSSION THREAD

**Discussion by mshearer0** [moj-analytical-services/splink#1612] [General]
I have a very simple model that trains over 3 iterations. 

On the first iteration the Probability two random records match is given as 0.0015 with an equivalent match weight of -13.2876.

![image](https://github.com/moj-analytical-services/splink/assets/37480681/ea33a1ca-36cd-4d61-97b0-59e5480b1531)

After the final iteration the probability is unchanged but the equivalent match weight is -9.3714 (which seems correct to me)

![image](https://github.com/moj-analytical-services/splink/assets/37480681/51b9d41b-a5f0-408c-b663-fbecf8847ef7)

The iteration 3 value of -9.3714 is displayed in the weights chart but the iteration 0 value of -13.2876 is used in the predictions and shown in the waterfall chart.

Is there something i'm not understanding here?

**Comment by RobinL** [comment]
I think this is probably because by default when you train a model using expectation maximization it only trains the m values.

Best parameter is allowed to vary during the training session but only for this stability of convergence, by default it isn't saved back to the model

See the arguments here:
https://moj-analytical-services.github.io/splink/linker.html?h=linker.estimate_parameters_using_expectation_maximisation#splink.linker.Linker.estimate_parameters_using_expectation_maximisation


Part of the rationale is probably here:
https://github.com/moj-analytical-services/splink/issues/462

**Comment by RobinL** [comment]
@mshearer0 aha, i have just run an example to understand this better - i think this is a bug, thanks for spotting it 

specifically
this line here: [_settings_obj](https://github.com/moj-analytical-services/splink/blob/13983df2305787232a3d71af680a11cb445ceaf7/splink/em_training_session.py#L289)

I believe should be:

```
for r in records:
                r["iteration"] = iteration
                r[
                    "probability_two_random_records_match"
                ] = settings_obj._probability_two_random_records_match
```

I will convert this to an issue and we will fix in due course
